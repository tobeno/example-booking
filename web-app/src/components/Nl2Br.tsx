import * as React from "react";

const Nl2Br: React.FC<{ text: string }> = ({ text }) => {
  const elements: Array<React.ReactNode> = [];

  let parts = text.split("\n");
  parts.forEach((part: string, index: number): void => {
    elements.push(<React.Fragment key={index}>{part}</React.Fragment>);

    if (index < parts.length - 1) {
      elements.push(<br key={`${index}-nl`} />);
    }
  });

  return <>{elements}</>;
};

export default Nl2Br;
