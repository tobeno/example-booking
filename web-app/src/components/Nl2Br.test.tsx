import { render } from "@testing-library/react";
import React from "react";
import Nl2Br from "./Nl2Br";

describe("Nl2Br", (): void => {
  it("renders text without newlines", (): void => {
    const { container } = render(<Nl2Br text="A nice text" />);
    expect(container).toMatchSnapshot();
  });

  it("renders text with newlines", (): void => {
    const { container } = render(
      <Nl2Br text={"A nice text\nwith even more text"} />,
    );
    expect(container).toMatchSnapshot();
  });
});
