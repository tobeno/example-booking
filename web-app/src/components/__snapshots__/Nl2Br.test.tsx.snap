// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nl2Br renders text with newlines 1`] = `
<div>
  A nice text
  <br />
  with even more text
</div>
`;

exports[`Nl2Br renders text without newlines 1`] = `
<div>
  A nice text
</div>
`;
